src/libcore/ptr/mod.rs

```diff
@@ -1266,7 +1266,7 @@ impl<T: ?Sized> *const T {
     ///
     /// # Panics
     ///
-    /// This function panics if `T` is a Zero-Sized Type ("ZST").
+    /// This function panics if `T` is a Zero-Sized Type ("ZST") or larger than `isize::max_value()`
```

Comment:
Such types are invalid in Rust anyway -- I don't think you can even state them?

Reply:
Well... the check is in there, So I thought it prudent to document it

Reply:
The check can never fail, though.
I am fine keeping the check as a fail-safe, but there is no reason to bother the user with this.